server/shared/src/main/scala/org/http4s/server/middleware/Throttle.scala

```diff
@@ -53,40 +56,50 @@ object Throttle {
       def takeToken: G[TokenAvailability] = fk(t.takeToken)
     }
 
+    private def getTime[F[_]: Temporal] =
+      Temporal[F].monotonic.map(_.toNanos)
+
     /** Creates an in-memory [[TokenBucket]].
       *
-      * @param capacity the number of tokens the bucket can hold and starts with.
+      * @param capacity    the number of tokens the bucket can hold and starts with.
       * @param refillEvery the frequency with which to add another token if there is capacity spare.
       * @return A task to create the [[TokenBucket]].
       */
     def local[F[_]](capacity: Int, refillEvery: FiniteDuration)(implicit
         F: Temporal[F]
-    ): F[TokenBucket[F]] = {
-      def getTime = F.monotonic.map(_.toNanos)
-      val bucket = getTime.flatMap(time => F.ref((capacity.toDouble, time)))
-
-      bucket.map { counter =>
-        new TokenBucket[F] {
-          override def takeToken: F[TokenAvailability] =
-            counter.access.flatMap { case ((previousTokens, previousTime), setter) =>
-              getTime.flatMap { currentTime =>
-                val timeDifference = currentTime - previousTime
-                val tokensToAdd = timeDifference.toDouble / refillEvery.toNanos.toDouble
-                val newTokenTotal = Math.min(previousTokens + tokensToAdd, capacity.toDouble)
-                def unavailable = {
-                  val timeToNextToken = refillEvery.toNanos - timeDifference
-                  TokenUnavailable(timeToNextToken.nanos.some)
-                }
-
-                // If setter fails (yields the value false), then retry with recursive call
-                if (newTokenTotal >= 1)
-                  setter((newTokenTotal - 1, currentTime)).ifM(F.pure(TokenAvailable), takeToken)
-                else
-                  setter((newTokenTotal, currentTime)).ifM(F.pure(unavailable), takeToken)
+    ): F[TokenBucket[F]] = for {
+      // toNanos throws if called on an infinite duration
+      nanoValue <- F.catchNonFatal(refillEvery.toNanos)
```

Comment:
`refillEvery: FiniteDuration` so that shouldn't happen :)

Reply:
Where did I see it as `Duration`? 😂
Will remove